### PR TITLE
edited MeanResponseTransformer to convert category to float, fixed tests

### PR DESCRIPTION
**Pull request by limlam96**
Fixed issue where category dtypes were preserved by this transformer, which is not desired behaviour. Edited to coerce outputs to float, as converting to numeric is ultimately the goal of this one.

Fixed tests, and edited some to put a category column through as part of testing.

### CODE REVIEW

File: tests/nominal/test_MeanResponseTransformer.py [hunk 1]
```diff
@@ -1112,17 +1135,30 @@ def test_expected_output_binary_response(self, df, expected):
     )
     def test_expected_output_one_multi_level(self, df, expected):
         """Test that the output is expected from transform with a multi-level response and one level selected."""
-        x = MeanResponseTransformer(columns=["b", "f"], level=["blue"])
+        columns = ["b", "f"]
+        level = ["blue"]
+        x = MeanResponseTransformer(columns=columns, level=level)
+
+        for col in [
```
Comment (davidhopkinson26) [inline tests/nominal/test_MeanResponseTransformer.py:1142]: could we move generation of this columns list to top of test and assign to an expanatory var name?  might make the test easier to read.  Same for other tests where you use this prefix + suffix format to generate column lists to iterate over

**Review by davidhopkinson26 — COMMENTED** [review]
Thanks for the contribution @limlam96! Idea and implementation look sound. This transformer is already a bit overcomplicated resulting in some difficult to interpret code and tests so most of my comments relate to avoiding making tests harder to read.

File: tests/nominal/test_MeanResponseTransformer.py (hunk 1 above)
Comment (limlam96) [reply tests/nominal/test_MeanResponseTransformer.py]: good suggestion, tidied these up here https://github.com/lvgig/tubular/pull/174/commits

File: tests/nominal/test_MeanResponseTransformer.py [hunk 2]
```diff
@@ -199,10 +214,13 @@ def test_output1(self):
         assert_series_equal(expected, output)
 
     def test_output2(self):
-        "Test output of method."
+        "Test output of method - for category dtype"
         x = MeanResponseTransformer(columns="a", prior=0)
 
-        x.fit(X=pd.DataFrame({"a": [1, 2]}), y=pd.Series([2, 3]))
+        df = pd.DataFrame({"a": ["a", "b"]})
+        df["a"] = df["a"].astype("category")
```
[the remark this answers is not in the document]
Comment (limlam96) [reply tests/nominal/test_MeanResponseTransformer.py]: fixed here https://github.com/lvgig/tubular/pull/174/commits

File: tests/nominal/test_MeanResponseTransformer.py [hunk 3]
```diff
@@ -199,10 +214,13 @@ def test_output1(self):
...
+
+        x.fit(X=df, y=pd.Series([2, 3]))
 
         expected1 = (1 * 1) / (1)
```
[the remark this answers is not in the document]
Comment (limlam96) [reply tests/nominal/test_MeanResponseTransformer.py:264]: as per out convo, fixed this and added func to control the return type 

https://github.com/lvgig/tubular/pull/174/commits/90a6e209721105d80c195fdcf9acd40d22741110

apologies this turned into a bigger one

File: tests/nominal/test_MeanResponseTransformer.py [hunk 4]
```diff
@@ -11,13 +13,26 @@
...
+    full_dict = {}
+
+    b_dict = {
         "b": {"a": 1.0, "b": 2.0, "c": 3.0, "d": 4.0, "e": 5.0, "f": 6.0},
         "b_blue": {"a": 1.0, "b": 1.0, "c": 0.0, "d": 0.0, "e": 0.0, "f": 0.0},
         "b_yellow": {"a": 0.0, "b": 0.0, "c": 1.0, "d": 1.0, "e": 0.0, "f": 0.0},
         "b_green": {"a": 0.0, "b": 0.0, "c": 0.0, "d": 0.0, "e": 1.0, "f": 1.0},
     }
 
+    # c matches b, but is categorical
```
[the remark this answers is not in the document]
Comment (limlam96) [reply tests/nominal/test_MeanResponseTransformer.py]: this is one where the nesting is confusing, have added a comment that hopefully helps

https://github.com/lvgig/tubular/pull/174/commits

but let me know if you have other thoughts

File: tests/nominal/test_MeanResponseTransformer.py [hunk 5]
```diff
@@ -364,15 +383,16 @@ def test_correct_mappings_stored(
         if level:
             if level == "all":
                 assert set(x.mapped_columns) == {
-                    "b_blue",
-                    "b_yellow",
-                    "b_green",
+                    prefix + "_" + suffix
```
[the remark this answers is not in the document]
Comment (limlam96) [reply tests/nominal/test_MeanResponseTransformer.py]: reworked this here https://github.com/lvgig/tubular/pull/174/commits

File: CHANGELOG.rst [hunk 6]
```diff
@@ -22,9 +22,9 @@ Each individual change should have a link to the pull request after the descript
 Added
```
Comment (davidhopkinson26) [inline CHANGELOG.rst:22]: please could you update the changelog version number and release date (to today)

File: tests/nominal/test_MeanResponseTransformer.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (limlam96) [reply tests/nominal/test_MeanResponseTransformer.py:259]: forgot to address the original comment.. so it turns out the to_dict call in _fit_binary_response converts types, so the type coercion makes sense to apply after this. This means I've left prior_encoder doing what it wants with types, so have left the type check aspect out of this test 

https://github.com/lvgig/tubular/pull/174/commits

File: CHANGELOG.rst [hunk 7]
```diff
@@ -22,9 +22,9 @@ Each individual change should have a link to the pull request after the descript
...
 ^^^^^
 - Updated NullIndicator to return int8 columns `#173 https://github.com/lvgig/tubular/pull/173`_
+- Update OneHotEncodingTransformer to default to returning int8 columns `#175 <https://github.com/lvgig/tubular/pull/175>`_
+- Updated MeanResponseTransformer to coerce return to float (useful behaviour for category type features) `#174 <https://github.com/lvgig/tubular/pull/174>`_
 
-1.1.1 (2024-01-18)
```
Comment (davidhopkinson26) [inline CHANGELOG.rst:26]: We should not have deleted this line

Comment (limlam96) [reply CHANGELOG.rst:26]: good catch, fixed here https://github.com/lvgig/tubular/pull/174/commits/5a5815546e34348b32cb7d605294dbd9ee979c78

File: CHANGELOG.rst (hunk 6 above)
Comment (limlam96) [reply CHANGELOG.rst:22]: updated date here https://github.com/lvgig/tubular/pull/174/commits/5a5815546e34348b32cb7d605294dbd9ee979c78

and version here
https://github.com/lvgig/tubular/pull/174/commits/b10649399f759695c69f7b9cf602ef825c486c67

File: tests/nominal/test_MeanResponseTransformer.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (davidhopkinson26) [reply tests/nominal/test_MeanResponseTransformer.py]: that is clearer thankyou

**Review by davidhopkinson26 — APPROVED** [review]
Happy with changes, thanks!

**Pull request merged**